Quantum-Control-Applications/Superconducting/Single Fixed Transmon/readout_duration_opt.py

```diff
@@ -0,0 +1,221 @@
+        IQ_st.buffer(2 * number_of_divisions).average().save("IQ")
+        QI_st.buffer(2 * number_of_divisions).average().save("QI")
+        QQ_st.buffer(2 * number_of_divisions).average().save("QQ")
+        # variances
+        (
+            ((II_st.buffer(2 * number_of_divisions) * II_st.buffer(2 * number_of_divisions)).average())
+            - (II_st.buffer(2 * number_of_divisions).average() * II_st.buffer(2 * number_of_divisions).average())
+        ).save("II_var")
+        (
+            ((IQ_st.buffer(2 * number_of_divisions) * IQ_st.buffer(2 * number_of_divisions)).average())
+            - (IQ_st.buffer(2 * number_of_divisions).average() * IQ_st.buffer(2 * number_of_divisions).average())
+        ).save("IQ_var")
+        (
+            ((QI_st.buffer(2 * number_of_divisions) * QI_st.buffer(2 * number_of_divisions)).average())
+            - (QI_st.buffer(2 * number_of_divisions).average() * QI_st.buffer(2 * number_of_divisions).average())
+        ).save("QI_var")
+        (
+            ((QQ_st.buffer(2 * number_of_divisions) * QQ_st.buffer(2 * number_of_divisions)).average())
+            - (QQ_st.buffer(2 * number_of_divisions).average() * QQ_st.buffer(2 * number_of_divisions).average())
+        ).save("QQ_var")
+
+#####################################
+#  Open Communication with the QOP  #
+#####################################
+qmm = QuantumMachinesManager(qop_ip)
+
+#######################
+# Simulate or execute #
+#######################
+
+simulate = True
+
+if simulate:
+    simulation_config = SimulationConfig(duration=1000)  # in clock cycles
+    job = qmm.simulate(config, opt_weights, simulation_config)
+    job.get_simulated_samples().con1.plot()
+else:
+    qm = qmm.open_qm(config)
+
+    job = qm.execute(opt_weights)
+    # Get results from QUA program
+    results = fetching_tool(job, data_list=["iteration"], mode="live")
+    # Live plotting
+    while results.is_processing():
+        # Fetch results
+        iteration = results.fetch_all()
+        # Progress bar
+        progress_counter(iteration, n_avg, start_time=results.get_start_time())
+
+    res_handles = job.result_handles
+
+    IIe, IIg = divide_array_in_half(res_handles.get("II").fetch_all())
+    IQe, IQg = divide_array_in_half(res_handles.get("IQ").fetch_all())
+    QIe, QIg = divide_array_in_half(res_handles.get("QI").fetch_all())
+    QQe, QQg = divide_array_in_half(res_handles.get("QQ").fetch_all())
+    IIe_var, IIg_var = divide_array_in_half(res_handles.get("II_var").fetch_all())
+    IQe_var, IQg_var = divide_array_in_half(res_handles.get("IQ_var").fetch_all())
+    QIe_var, QIg_var = divide_array_in_half(res_handles.get("QI_var").fetch_all())
+    QQe_var, QQg_var = divide_array_in_half(res_handles.get("QQ_var").fetch_all())
+
+    Ie = IIe + IQe
+    Ig = IIg + IQg
+    Qe = QIe + QQe
+    Qg = QIg + QQg
+    Ie_var = IIe_var + IQe_var
+    Ig_var = IQg_var + IIg_var
+    Qe_var = QIe_var + QQe_var
+    Qg_var = QQg_var + QIg_var
```

Comment:
I am not sure that this is mathematically correct since the variance is not linear...
I think that the correct way of writing it would be
`<(II+IQ)*(II+IQ)> - <II+IQ><II+IQ>` which is different from `<II*II>-<II>*<II> + <IQ*IQ>-<IQ>*<IQ>`
Note that I and Q can be calculated directly in QUA by doing
`assign(I, II+IQ)` & `assign(Q, QI+QQ)` before the save statements, what do you think?

Reply:
thanks Theo, corrected accordingly in code